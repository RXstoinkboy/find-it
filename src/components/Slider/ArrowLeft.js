import React from 'react';
import styles from './styles/ArrowLeft.module.scss'
import PropTypes from 'prop-types'

export const ArrowLeft = props => {
    return (
        <button className={styles.arrowLeft} onClick={props.prevSlide} />
    );
};

export default ArrowLeft;

ArrowLeft.propTypes = {
    prevSlide: PropTypes.func.isRequired
}