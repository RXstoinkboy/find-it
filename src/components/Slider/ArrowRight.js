import React from 'react';
import styles from './styles/ArrowRight.module.scss'
import PropTypes from 'prop-types'

export const ArrowRight = props => {
    return (
        <button className={styles.arrowRight} onClick={props.nextSlide} />
    );
};

export default ArrowRight;

ArrowRight.propTypes = {
    nextSlide: PropTypes.func.isRequired
}