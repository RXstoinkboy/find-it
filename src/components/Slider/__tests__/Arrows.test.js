import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import {ArrowRight} from '../ArrowRight'
import {ArrowLeft} from '../ArrowLeft'

describe(`Arrows`, ()=> {
    const prevSlideMock = jest.fn()
    const nextSlideMock = jest.fn()

    it(`render correctly`, ()=> {
        const right = shallow(<ArrowRight onClick={nextSlideMock} nextSlide={nextSlideMock} />);
        const left = shallow(<ArrowLeft onClick={prevSlideMock} prevSlide={prevSlideMock} />);

        expect(toJson(right)).toMatchSnapshot();
        expect(toJson(left)).toMatchSnapshot();
    })
})