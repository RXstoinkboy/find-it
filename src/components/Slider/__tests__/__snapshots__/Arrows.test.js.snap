// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrows render correctly 1`] = `
<button
  className="arrowRight"
  onClick={[MockFunction]}
/>
`;

exports[`Arrows render correctly 2`] = `
<button
  className="arrowLeft"
  onClick={[MockFunction]}
/>
`;
